packages/layers/auth/tests/unit/composables/useAuthApi.test.ts

```diff
@@ -0,0 +1,149 @@
+  })
+})
+
+const mockOpenErrorModal = vi.fn()
+mockNuxtImport('useConnectTosModals', () => {
+  return () => ({
+    openPatchTosErrorModal: mockOpenErrorModal
+  })
+})
+
+const mockQuery = vi.fn()
+const mockInvalidateQueries = vi.fn()
+
+const { mockDefineQuery, mockUseMutation } = vi.hoisted(() => {
+  const mockMutationReturnObject = {
+    mutateAsync: vi.fn(),
+    isPending: false,
+    isSuccess: false
+  }
+
+  const mockUseMutation = vi.fn(() => mockMutationReturnObject)
+
+  const mockDefineQuery = vi.fn(() => mockQuery)
+
+  return {
+    mockDefineQuery,
+    mockUseMutation
+  }
+})
+
+vi.mock('@pinia/colada', () => ({
+  defineQuery: mockDefineQuery,
+  useQueryCache: () => ({ invalidateQueries: mockInvalidateQueries }),
+  useMutation: mockUseMutation,
+  defineMutation: vi.fn(setupFn => setupFn)
+}))
+
+describe('useAuthApi', () => {
+  const authApi = useAuthApi()
+
+  beforeEach(() => {
+    vi.clearAllMocks()
+  })
+
+  describe('getAuthUserProfile', () => {
+    it('should define and return the query', async () => {
+      const expected = { data: 'test-data' }
+      mockQuery.mockReturnValue(expected)
+
+      const result = await authApi.getAuthUserProfile()
+
+      expect(mockDefineQuery).toHaveBeenCalledTimes(1)
+
+      const callArgs = (mockDefineQuery.mock.calls[0] as any)[0]
+
+      expect(callArgs.key).toEqual(['auth-user-profile'])
+      expect(callArgs.staleTime).toBe(300000)
+
+      expect(mockQuery).toHaveBeenCalledTimes(1)
+
+      expect(result).toBe(expected)
+
+      const queryFunction = callArgs.query
+      queryFunction()
+
+      expect(mockAuthApi).toHaveBeenCalledWith('/users/@me', {
+        parseResponse: JSON.parse
+      })
+    })
+  })
+
+  describe('getTermsOfUse', () => {
+    it('should define and return the query', async () => {
+      const expected = { data: 'test-data' }
+      mockQuery.mockReturnValue(expected)
+
+      const result = await authApi.getTermsOfUse()
+
+      expect(mockDefineQuery).toHaveBeenCalledTimes(1)
+
+      const callArgs = (mockDefineQuery.mock.calls[0] as any)[0]
+
+      expect(callArgs.key).toEqual(['auth-terms-of-use'])
+      expect(callArgs.staleTime).toBe(300000)
+
+      expect(mockQuery).toHaveBeenCalledTimes(1)
+
+      expect(result).toBe(expected)
+
+      const queryFunction = callArgs.query
+      queryFunction()
+
+      expect(mockAuthApi).toHaveBeenCalledWith('/documents/termsofuse')
+    })
+  })
+
+  describe('usePatchTermsOfUse', () => {
+    const mockSuccessCb = vi.fn(() => Promise.resolve())
+    const { usePatchTermsOfUse } = useAuthApi()
+    const mutateArgs = { accepted: true, version: '5', successCb: mockSuccessCb }
+
+    usePatchTermsOfUse()
+    const callArgs = (mockUseMutation.mock.calls[0] as any)[0]
+
+    beforeEach(() => {
+      vi.clearAllMocks()
+    })
+
+    it('should call authApi with correct payload and endpoint', async () => {
+      mockAuthApi.mockResolvedValue({})
+
+      await callArgs.mutation(mutateArgs)
```

Comment:
Could we call the 'patchTermsOfUse' from 'usePatchTermsOfUse' instead if we aren't mocking the mutation? And check the mockAuthApi is called correctly afterwards? I think that's a lot simpler and if we're integrated incorrectly with pinia colada it'll catch the error here instead of passing due to the mock